vpd-parser/memory_vpd_parser.cpp

```diff
@@ -36,72 +42,306 @@ static constexpr auto JEDEC_SDRAMCAP_RESERVED = 6;
 static constexpr auto JEDEC_RESERVED_BITS = 3;
 static constexpr auto JEDEC_DIE_COUNT_RIGHT_SHIFT = 4;
 
+static constexpr auto JEDEC_SDRAM_NUM_OF_CHNLS_PER_DIMM_MASK = 0x60;
+static constexpr auto JEDEC_PRI_BUS_WIDTH_PER_CHNL_MASK = 0x07;
+static constexpr auto JDEC_SDRAM_IO_WIDTH_MASK = 0xE0;
+static constexpr auto JDEC_SDRAM_DIE_PER_PKG_MASK = 0xE0;
+static constexpr auto JDEC_SDRAM_DENSITY_PER_DIE_MASK = 0x1F;
+static constexpr auto JDEC_NUM_OF_PKG_RANK_PER_CHNL_MASK = 0x38;
+static constexpr auto JDEC_SDRAM_RANK_MIX_TYPE_MASK = 0x40;
+
+static constexpr auto JEDEC_NUM_OF_CHNLS_PER_DIMM_COUNT_RIGHT_SHIFT = 5;
+static constexpr auto JEDEC_NUM_OF_PKG_RANK_PER_COUNT_RIGHT_SHIFT = 3;
+static constexpr auto JEDEC_SDRAM_IO_WIDTH_COUNT_RIGHT_SHIFT = 5;
+static constexpr auto JEDEC_SDRAM_DIE_PER_PKG_COUNT_RIGHT_SHIFT = 5;
+static constexpr auto JEDEC_SDRAM_RANK_MIX_TYPE_BIT_POSITION = 6;
+
+static constexpr auto JEDEC_NUM_OF_CHNL_PER_DIMM_RESERVED_VALUE = 1;
+static constexpr auto JEDEC_PRI_BUS_WIDTH_PER_CHNL_RESERVED_VALUE = 3;
+static constexpr auto JEDEC_NUM_OF_PKG_RANKS_PER_CHNL_RESERVED_VALUE = 7;
+static constexpr auto JEDEC_SDRAM_IO_WIDTH_RESERVED_VALUE = 3;
+static constexpr auto JEDEC_DIE_PER_PKG_RESERVED_VALUE1 = 1;
+static constexpr auto JEDEC_DIE_PER_PKG_RESERVED_VALUE2 = 5;
+static constexpr auto JEDEC_SDRAM_DENSITY_PER_DIE_RESERVED_VALUE = 8;
+
+static constexpr auto JEDEC_PRI_BUS_WIDTH_PER_CHNL_MULTIPLIER = 8;
+
+static const std::unordered_map<uint8_t, uint8_t> diePerPkgMap = {
+		{0, 1}, {2, 2}, {3, 4}, {4, 8}, {5, 16}};
+
+static const std::unordered_map<uint8_t, uint8_t> sdramDnstyPerDieMap = {
+		{0, 0}, {1, 4}, {2, 8}, {3, 12}, {4, 16}, {5, 24}, {6, 32}, {7, 48}, {8, 64}};
+
 auto memoryVpdParser::getDimmSize(Binary::const_iterator iterator)
 {
-    size_t tmp = 0, dimmSize = 0;
```

Comment:
need to keep DDr4 check too along with DDR5

Reply:
> need to keep DDr4 check too along with DDR5

DDR4 check is there, I have not removed, due to formatting issue you might have not found it. Please check now